### PR TITLE
Fix broken links to package manuals

### LINKED ISSUE #252: Links on https://www.gap-system.org/Doc/manuals.html all not working
Most links on https://www.gap-system.org/Doc/manuals.html for the package docs links give a 404 page. Removing the version number from the provided links makes them work. I am not sure what exactly leads to this problem, if they are misslinked o manual.html or if they are provided on the wrong url.

(E.g. https://www.gap-system.org/Manuals/pkg/images-1.3.0/doc/chap0.html is linked but https://www.gap-system.org/Manuals/pkg/images/doc/chap0.html works)

### PR DESCRIPTION
**Pull request by wilfwilson**
The links to manuals would use the `InstallationPath` value of a package for its directory name, but this sometimes includes the version number. However, the `extract_manuals.py` script now strips any version number from the directory name when extracting the package manuals for uploading to the website.

So we now need to remove any version numbers when making the links too. I do this by stripped anything after (and include) a hyphen, but this is risky if a package ever properly contains a hyphen as part of its name. Would it be safe to just use `dir = pkg[1].PackageName`?

Fixes #252.

Thanks to @veni-vidi-code for the report.

### CODE REVIEW

**Comment by fingolfin** [comment]
LOL we submitted PRs within 1 minute from each other 😂

**Comment by fingolfin** [comment]
Right now, this patch is fine, and matches what we do on the separate package pages.

Long term, though, we should instead adjust the `InstallationPath` values (as stored in `_data/package-infos/4-11-1.json`) to reflect the actual directory name. I.e, the `update_website.py` script should print that modified path into the JSON file. Then we can undo the patch here, and simplify `extract_manuals.py` and `_layouts/packages.html`

**Comment by wilfwilson** [comment]
I think we want to retain the value `InstallationPath`, because it gives the name of the directory that is actually contained in the archive e.g. `packages-v4.11.1.tar.gz`, as far as I am aware. So unless we are _also_ going to change the package-wrapping scripts to remove the version numbers from the directory names, then we should retain it. (Although that doesn't mean that we should have an additional key that stores the directory name with the path removed.

In summary, there are two different directory names: the one used by the GAP website for a package, and the one that exists in `packages-vX.Y.Z.tar.gz` (which has other uses beyond the website).

**Comment by fingolfin** [comment]
Ah, I wasn't aware anymore that we also use `InstallationPath` in other scripts. But of course we do.

So then I'd add an *additional* key with the "cleaned up" name. Or: we could indeed normalize the directory names, e.g. to "package name in lower case"; this would of course (again) break any links. (Although it would be possible to set up redirects with nginx...
